server/pxf-hbase/src/main/java/org/greenplum/pxf/plugins/hbase/HBaseAccessor.java

```
@@ -146,6 +146,14 @@ public void closeForRead() throws Exception {
         HBaseUtilities.closeConnection(null, connection);
     }
 
+    /**
+     * Cancel read operation.
+     */
+    @Override
+    public void cancelRead() {
+        throw new UnsupportedOperationException("Cancel read operation is not supported");
```

I think we should either do nothing or call currentScanner.close()

---

I think we should test whether currentScanner.close() will be enough to cancel request. And implement this method only after the test. Agree, we can do nothing instead of  throw `UnsupportedOperationException. Any case, It will not change the logic as we ignore the errors in the method ReadServiceImpl#cancelExecution:
```java
    private boolean cancelExecution(RequestIdentifier requestIdentifier, Bridge bridge) {
        if (bridge == null) {
            log.debug("Couldn't cancel read request, request {} not found", requestIdentifier);
            return false;
        }
        try {
            log.debug("Cancelling read request {}", requestIdentifier);
            bridge.cancelIteration();
        } catch (Exception e) {
            log.warn("Ignoring error encountered during bridge.cancelIteration()", e);
            return false;
        }
        return true;
    }
```

---

Yep, I've seen this part, but I don't think it'a a good idea to have log full of warnings

---

Yes, good point. Tacking into account the notes from @iamlapa about the additional interface I would suggest just to log the information without throwing the exception.  Paying attention that the customer might have a big amount of logical segments I would set the log level to debug level for `UnsupportedOperationException`